run.go

```diff
@@ -326,38 +329,96 @@ func Log(c chan *SimulationStatus) DomainManipulator {
+	for _, val := range s {
+		if _, ok := seen[val]; !ok {
+			result = append(result, val)
+			seen[val] = val
+		}
+	}
+	return result
+}
+
+// Checks if outputVariables are defined in terms of other outputVariables.
+// Creates a slice of unique model variables that are required to calculate
+// the requested outputVariables.
+func checkForDerivitives(m map[string]string) (map[string]string, []string) {
+	getVariables := []string{}
+	for key, val := range m {
+		expression, _ := govaluate.NewEvaluableExpression(val)
+		uniqueVars := removeDuplicates(expression.Vars())
+		getVariables = append(getVariables, uniqueVars...)
+		for _, uniqueVar := range uniqueVars {
+			if m[uniqueVar] != "" && m[uniqueVar] != uniqueVar {
+				m[key] = strings.Replace(m[key], uniqueVar, "("+m[uniqueVar]+")", -1)
+				return checkForDerivitives(m)
+			}
+		}
+	}
+	return m, removeDuplicates(getVariables)
+}
+
+func (d *InMAP) checkGetNames(g ...string) error {
 	tempOutputNames, _, _ := d.OutputOptions()
 	outputNames := make(map[string]uint8)
 	for _, n := range tempOutputNames {
 		outputNames[n] = 0
 	}
-	for _, v := range n {
+	for _, v := range g {
 		if _, ok := outputNames[v]; !ok {
 			return fmt.Errorf("inmap: unsupported output variable name '%s'", v)
 		}
 	}
 	return nil
 }
 
+func (d *InMAP) checkOutputNames(o map[string]string) error {
+	for key, _ := range o {
+		if len(key) > 10 {
+			return fmt.Errorf("inmap: output variable name '%s' exceeds 10 characters", o)
+		}
+	}
+	return nil
+}
+
 // Results returns the simulation results.
 // Output is in the form of map[variable][row]concentration.
 // If  allLayers` is true, the function returns data for all of the vertical
 // layers, otherwise only the ground-level layer is returned.
 // outputVariables is a list of the names of the variables for which data should be
 // returned.
-func (d *InMAP) Results(allLayers bool, outputVariables ...string) (map[string][]float64, error) {
-	if err := d.checkOutputNames(outputVariables...); err != nil {
+func (d *InMAP) Results(allLayers bool, outputVariables map[string]string) (map[string][]float64, error) {
+
+	outputVariables, getVariables := checkForDerivitives(outputVariables)
+
+	if err := d.checkGetNames(getVariables...); err != nil {
+		return nil, err
+	}
+
+	if err := d.checkOutputNames(outputVariables); err != nil {
 		return nil, err
 	}
 
 	// Prepare output data
+	getConc := make(map[string][]float64)
+	eachRowConc := make(map[string]interface{})
 	outputConc := make(map[string][]float64)
-	for _, name := range outputVariables {
+	for _, name := range getVariables {
 		if allLayers {
-			outputConc[name] = d.toArray(name, -1)
+			getConc[name] = d.toArray(name, -1)
 		} else {
-			outputConc[name] = d.toArray(name, 0)
+			getConc[name] = d.toArray(name, 0)
+		}
+	}
+	for k, v := range outputVariables {
+		arrayConc := v
+		for i, v2 := range arrayConc {
+			eachRowConc[k] = v2
+			expression, _ := govaluate.NewEvaluableExpression(v)
```

Comment:
Whereever there are errors being returned, the should be dealt with rather than ignored.